opm/autodiff/BlackoilPropsAdFromDeck.cpp

```diff
@@ -928,5 +928,42 @@ BlackoilPropsAdFromDeck::BlackoilPropsAdFromDeck(const BlackoilPropsAdFromDeck&
     }
 
 
+    /// Obtain the scaled critical oil in gas saturation values.
+    /// \param[in]  cells  Array of cell indices.
+    /// \return Array of critical oil in gas saturaion values.
+    V BlackoilPropsAdFromDeck::scaledCriticalOilinGasSaturations(const Cells& cells) const {
+
+
+        assert(phaseUsage_.phase_used[BlackoilPhases::Gas]);
+        assert(phaseUsage_.phase_used[BlackoilPhases::Oil]);
+
+        const int n = cells.size();
+        V sgocr = V::Zero(n);
+        for (int i = 0; i < n; ++i) {
+            const auto& scaledDrainageInfo =
+                satprops_->materialLawManager()->oilWaterScaledEpsInfoDrainage(cells[i]);
```

Comment:
This will be quite slow: a shared pointer is constructed in an inner loop. If the method materialLawManager() instead returns a const reference it becomes better, but it would probably be better to put the call outside the loop anyway:
```
const MaterialLawManager& mlaw = satprops_->materialLawManager();
for (...) {
    // Use mlaw.oilWaterScaledEpsInfoDrainage().
    ...
}
```